src/marqo/tensor_search/add_docs.py

```diff
@@ -88,10 +88,10 @@ def threaded_download_and_preprocess_content(allocated_docs: List[dict],
                 if field not in tensor_fields:
```

Comment:
@farshidz 
There are two ways to solve the issue when users provide an incorrect modality for a structured index:
- We infer the modality and see if there is a mismatch, which is what we are doing in this PR
- We don’t infer the modality and just process it as if it is the expected content and catch the raised error

Option 1 is more effective, but we need to be confident with our inferred modality.
Option 2 is more straightforward but it is more complicated and slower because we need to download the content, process it, and catch the error.